File: DSA/bfa.py
"""Naive patterntern matching algorithm
        1)Define Pattern and the test string
        2) traverse through the pattern and the test string ( i is test string, j is pattern string)
        3) if the character is matching then increment i and j
        4) if not increment i and backtrack j
"""
def search(pattern, test):
    M = len(pattern)
    N = len(test)
 
    # A loop to slide pat[] one by one */
    for i in range(N - M + 1):
        j = 0
 
        # For current index i, checkujikol p;                                                                        
        # for pattern match */
        while(j < M):
            if (test[i + j] != pattern[j]):
                break
            j += 1
 
        if (j == M):
            print("Pattern found at index ", i)
 
 

test = "AABAACAADAABAAABAA"
pattern = "AABA"
     
# Function call
search(pattern, test)

File: DSA/bfa.c
#include <stdio.h>
#include <string.h>
 
void search(char* pat, char* txt)
{
    int M = strlen(pat);
    int N = strlen(txt);
 
   /*outer loop is for moving the index one by one in the string */
    for (int i = 0; i < N - M+1; i++) {
        int j;
        /* The inner loop is for checking  */
        for (int j = 0; j < M; j++)
            if (txt[i + j] != pat[j])
                break;
 
        if (j== M) // if pat[0...M-1] = txt[i, i+1, ...i+M-1]
            printf("Pattern found at index %d \n", i);
    }
}
 
int main()
{
    char txt[] = "Jayant Toleti";
    char pat[] = "i";
   
      // Function call
    search(pat, txt);
    return 0;
}